app/modals/stacking/stacking-modal.tsx

```diff
@@ -56,47 +71,96 @@ export const StackingModal: FC<StackingModalProps> = ({ onClose }) => {
       ? StackingModalStep.DecryptWalletAndSend
       : StackingModalStep.SignWithLedgerAndSend;
   const [step, setStep] = useState(initialStep);
 
-  const createSoftwareWalletTx = useCallback(async () => {
-    if (!password || !encryptedMnemonic || !salt) {
+  const createSoftwareWalletTx = useCallback(async (): Promise<StacksTransaction> => {
+    if (!password || !encryptedMnemonic || !salt || !poxInfo || !balance) {
       throw new Error('One of `password`, `encryptedMnemonic` or `salt` is missing');
     }
+    const poxClient = new POX(node.url);
     const { privateKey } = await decryptSoftwareWallet({
       ciphertextMnemonic: encryptedMnemonic,
       salt,
       password,
     });
-    console.log({ privateKey });
-    // return makeSTXTokenTransfer({ ...options, senderKey: privateKey });
-  }, [encryptedMnemonic, password, salt]);
+    const balanceBN = new BigNumber(balance, 10);
+    const txOptions = poxClient.getLockTxOptions({
+      cycles: numCycles,
+      poxAddress,
+      amountMicroSTX: balanceBN,
+      contract: poxInfo.contract_id,
+    });
+    const tx = await makeContractCall({
+      ...txOptions,
+      senderKey: privateKey,
+    });
+    poxClient.modifyLockTxFee({ tx, amountMicroSTX: balanceBN });
+    const signer = new TransactionSigner(tx);
+    signer.signOrigin(createStacksPrivateKey(privateKey));
+    return tx;
+  }, [encryptedMnemonic, password, salt, numCycles, balance, poxInfo, poxAddress, node.url]);
 
   const createLedgerWalletTx = useCallback(
-    async (options: { publicKey: Buffer }) => {
-      console.log(options);
-      if (!publicKey || !blockstackApp)
-        throw new Error('`publicKey` or `blockstackApp` is not defined');
+    async (options: { publicKey: Buffer }): Promise<StacksTransaction> => {
+      const poxClient = new POX(node.url);
+      if (!blockstackApp || !poxInfo || !balance)
+        throw new Error('`poxInfo` or `blockstackApp` is not defined');
       // 1. Form unsigned contract call transaction
+      const balanceBN = new BigNumber(balance, 10);
+      const txOptions = poxClient.getLockTxOptions({
+        amountMicroSTX: balanceBN,
+        poxAddress,
+        cycles: numCycles,
+        contract: poxInfo.contract_id,
+      });
+
+      const unsignedTx = await makeUnsignedContractCall({
+        ...txOptions,
+        publicKey: options.publicKey.toString('hex'),
+      });
+
+      poxClient.modifyLockTxFee({ tx: unsignedTx, amountMicroSTX: balanceBN });
 
       // 2. Sign transaction
-      // const resp = await blockstackApp.sign(`m/44'/5757'/0'/0/0`, unsignedTx.serialize());
+      const resp: ResponseSign = await blockstackApp.sign(
+        `m/44'/5757'/0'/0/0`,
+        unsignedTx.serialize()
+      );
+      if (resp.returnCode !== LedgerError.NoErrors) {
+        throw new Error('Ledger responded with errors');
+      }
 
-      // 3. Add signature to unsigned tx
+      unsignedTx.createTxWithSignature(resp.signatureVRS);
```

Comment:
This call returns a new tx without mutating the original.

```ts
const signedTx = unsignedTx.createTxWithSignature(resp.signatureVRS);
```

Reply:
Thanks, fixed. If it's immutable, then I think `createTxWithSignature` makes more sense.